### PR TITLE
Device Datasource

### PR DESCRIPTION
**Pull request by rounak-adhikary**
# Description
Add device datasource

# GitHub Issues

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE
Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
ome_device

##### OUTPUT

image

image

##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by Krishnan-Priyanshu — APPROVED** [review]
LGTM

File: examples/data-sources/ome_device/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  filters = {
+    ids = [1001, 1002, 1003]
+  }
+}
+
+# get device by their network
+data "ome_device" "devn" {
+  filters = {
+    ip_expressions = [
+      "10.226.197.136",
```
Comment (Krunal-Thakkar) [inline examples/data-sources/ome_device/data-source.tf:25]: Please remove actual ips

Comment (rounak-adhikary) [reply examples/data-sources/ome_device/data-source.tf]: done

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**